### DISCUSSION TITLE
Freeze individual parameters

### DISCUSSION THREAD

**Discussion by Jaznos** [JSBSim-Team/jsbsim#1085] [Q&A]
Hello!
is there  any idea of how to freeze single parameter like the heading or the pitch? i know that there is a sort of position freeze and attitude freeze implimented that consiste of avoiding the integration of the angular and linear positions and rates
thanks!

**Comment by seanmcleod** [comment]
I'm not aware of any ability to freeze individual state.

Maybe you're thinking of `SuspendIntegration()`? Although it's not specific to position and attitude. It's used by code like the trim code etc. 

https://github.com/JSBSim-Team/jsbsim/blob/56c666265ac1db3216cc76c7980a87ec424a6546/src/FGFDMExec.h#L554-L555

There is also a hold option, but again it's global.

https://github.com/JSBSim-Team/jsbsim/blob/56c666265ac1db3216cc76c7980a87ec424a6546/src/FGFDMExec.h#L487-L488

**Comment by Jaznos** [comment]
i am talking about this part on FGPropagate.cpp
using eNone as integrator_rotational_position and  integrator_rotational_rate to freeze attitude
and integrator_translational_position and integrator_translational_rate to freeze position


```c++
 if (!FDMExec->IntegrationSuspended()) {
    Integrate(VState.qAttitudeECI,      VState.vQtrndot,      VState.dqQtrndot,          dt, integrator_rotational_position);
    Integrate(VState.vPQRi,             in.vPQRidot,          VState.dqPQRidot,          dt, integrator_rotational_rate);
    Integrate(VState.vInertialPosition, VState.vInertialVelocity, VState.dqInertialVelocity, dt, integrator_translational_position);
    Integrate(VState.vInertialVelocity, in.vUVWidot,          VState.dqUVWidot,          dt, integrator_translational_rate);
  }
```

**Comment by Jaznos** [comment]
i think that the probleme is that the integration s are done in the inertial frame
if we were on the local frame we have juste to set the same value each iteration for exemple give the same value to the heading each iteration
VState.qAttitudeLocal(Psi) =same value.

**Comment by bcoconni** [comment]
As you mentioned JSBSim is not able to freeze an individual parameter and there are currently no plans to do so. Also note that this kind of feature may not result in what most people are expecting: for example, freezing the heading in the local frame does not result in the aircraft flying along a geodesic (i.e. a "straight" line) unless its heading is exactly 0, 90, 180 or 270 degrees.
